presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -199,13 +199,13 @@ public static boolean isConstant(Expression expression)
 
         // ROW an MAP are special so we explicitly do that here.
         if (tempExpression instanceof Row) {
-            return !(((Row) tempExpression).getItems().stream().filter(x -> !isConstant(expression)).findAny().isPresent());
+            return !(((Row) tempExpression).getItems().stream().filter(x -> !isConstant(x)).findAny().isPresent());
```

Comment:
this is why it's probably better to use `filter(ExpressionTreeUtils::isConstant)` instead.

Reply:
Done

allMatch(ExpressionTreeUtils::isConstant)